csharp/client/DeephavenClient/ArrowTable.cs

```diff
@@ -0,0 +1,198 @@
+    status1.OkOrThrow();
+
+    var columnHandles = new StringHandle[NumColumns];
+    var elementTypesAsInt = new Int32[NumColumns];
+    NativeArrowTable.deephaven_client_ArrowTable_GetSchema(self, NumColumns, columnHandles, elementTypesAsInt, out var stringPoolHandle, out var status2);
+    status2.OkOrThrow();
+    var pool = stringPoolHandle.ExportAndDestroy();
+
+    _columnNames = new string[NumColumns];
+    _columnElementTypes = new ElementTypeId[NumColumns];
+    for (var i = 0; i != NumColumns; ++i) {
+      _columnNames[i] = pool.Get(columnHandles[i]);
+      _columnElementTypes[i] = (ElementTypeId)elementTypesAsInt[i];
+    }
+  }
+
+  ~ArrowTable() {
+    ReleaseUnmanagedResources();
+  }
+
+  public void Dispose() {
+    ReleaseUnmanagedResources();
+    GC.SuppressFinalize(this);
+  }
+
+  public (Array, bool[]) GetColumn(Int32 index) {
+    var elementType = _columnElementTypes[index];
+    var factory = ArrowTableColumnFactory.Of(elementType);
+    var (data, nulls) = factory.GetColumn(Self, index, NumRows);
+    return (data, nulls);
+  }
+
+  public Array GetNullableColumn(Int32 index) {
+    var elementType = _columnElementTypes[index];
+    var factory = ArrowTableColumnFactory.Of(elementType);
+    return factory.GetNullableColumn(Self, index, NumRows);
+  }
+
+  public void ReleaseUnmanagedResources() {
+    if (!Self.TryRelease(out var old)) {
+      return;
+    }
+    NativeArrowTable.deephaven_client_ArrowTable_dtor(old);
+  }
+}
+
+internal static class ArrowTableColumnFactory {
+  private static readonly ColumnFactory<NativeArrowTable>[] Factories = {
+    new ColumnFactory<NativeArrowTable>.ForChar(NativeArrowTable.deephaven_client_ArrowTable_GetCharAsInt16Column),
+    new ColumnFactory<NativeArrowTable>.ForOtherValueType<SByte>(NativeArrowTable.deephaven_client_ArrowTable_GetInt8Column),
+    new ColumnFactory<NativeArrowTable>.ForOtherValueType<Int16>(NativeArrowTable.deephaven_client_ArrowTable_GetInt16Column),
+    new ColumnFactory<NativeArrowTable>.ForOtherValueType<Int32>(NativeArrowTable.deephaven_client_ArrowTable_GetInt32Column),
+    new ColumnFactory<NativeArrowTable>.ForOtherValueType<Int64>(NativeArrowTable.deephaven_client_ArrowTable_GetInt64Column),
+    new ColumnFactory<NativeArrowTable>.ForOtherValueType<float>(NativeArrowTable.deephaven_client_ArrowTable_GetFloatColumn),
+    new ColumnFactory<NativeArrowTable>.ForOtherValueType<double>(NativeArrowTable.deephaven_client_ArrowTable_GetDoubleColumn),
+    new ColumnFactory<NativeArrowTable>.ForBool(NativeArrowTable.deephaven_client_ArrowTable_GetBooleanAsInteropBoolColumn),
+    new ColumnFactory<NativeArrowTable>.ForString(NativeArrowTable.deephaven_client_ArrowTable_GetStringColumn),
+    new ColumnFactory<NativeArrowTable>.ForDateTime(NativeArrowTable.deephaven_client_ArrowTable_GetDateTimeAsInt64Column),
+    // List - TODO(kosak)

```

Comment:
Can you turn this "TODO" to a comment describing a potential future change?
Asking for (1) less cryptic (2) not framing as a TODO (TODO = "we should have but don't have yet").

Reply:
PTAL